src/Formats/JSONUtils.cpp

```diff
@@ -286,11 +291,78 @@ namespace JSONUtils
                 return true;
             }
 
-            if (as_nullable)
-                return SerializationNullable::deserializeNullAsDefaultOrNestedTextJSON(column, in, format_settings, serialization);
+            auto do_deserialize = [](IColumn & column_, ReadBuffer & buf_, auto && check_for_empty_string, auto && deserialize) -> bool
+            {
+                if (check_for_empty_string(buf_))
+                {
+                    column_.insertDefault();
+                    return false;
+                }
+                else
+                    return deserialize(column_, buf_);
+            };
+
+            auto deserialize_impl = [as_nullable, &format_settings, &serialization](IColumn & column_, ReadBuffer & buf_) -> bool
+            {
+                if (as_nullable)
+                    return SerializationNullable::deserializeNullAsDefaultOrNestedTextJSON(column_, buf_, format_settings, serialization);
+
+                serialization->deserializeTextJSON(column_, buf_, format_settings);
+                return true;
+            };
+
+            if (!format_settings.json.empty_as_default  || in.eof() || *in.position() != EMPTY_STRING[0])
```

Comment:
```suggestion
            if (!format_settings.json.empty_as_default || in.eof() || *in.position() != EMPTY_STRING[0])
```

Reply:
Done (I avoid using the script that formats automatically the source code since I noticed that it impacts a lot of files unrelated to my PR).